### PR TITLE
Linalg.norm ndim along axis partial fix

### PR DESCRIPTION
**Pull request by TAdeJong**
This is a partial fix for #3895, working towards feature parity with numpy's `linalg.norm`. 
For the implementation of SVD based norms along arbitrary axis the `tsqr` algorithm could be extended, or some clever choice of rechunk/map_blocks should be made, both of which I feel not confident to be able to fix at this point in time, as such the code will throw `NotImplementedErrors` for now in these cases.

I feel however that the extension of functionality for arrays of vectors and arrays of matrices for non-SVD based norms makes it still worth it to merge. If we merge, I will create a new issue for the SVD based norms.

Feedback is welcome as always.

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

File: dask/array/linalg.py [hunk 1]
```diff
@@ -111,6 +111,7 @@ def tsqr(data, compute_svd=False, _max_vchunk_size=None):
             "  2. Have only one column of blocks\n\n"
             "Note: This function (tsqr) supports QR decomposition in the case of\n"
             "tall-and-skinny matrices (single column chunk/block; see qr)"
+            "shape: {}, chunks: {}".format(data.shape, data.chunks)
```
Comment (mrocklin) [inline dask/array/linalg.py]: Note that `data.chunks` can be quite large in practice.  I wonder if `data.chunksize` would be more appropriate.

**Comment by TAdeJong** [comment]
@mrocklin: thanks for the feedback. I did not consider speed yet when I was writing these tests, and have plenty ideas to speed this up, both in simply tuning the test cases and some additional stuff. I will have a look in the near future.

**Comment by mrocklin** [comment]
Thanks!

Keeping tests fast helps encourage people to run them frequently when
developing :)

File: dask/array/linalg.py [hunk 2]
```diff
@@ -1081,16 +1082,16 @@ def lstsq(a, b):
...
         axis = (int(axis),)
     else:
         axis = tuple(axis)
 
+    if len(axis) > 2:
+        raise ValueError("Improper number of dimensions to norm.")
+
     if ord == "fro":
         ord = None
         if len(axis) == 1:
```
Comment (jakirkham) [inline dask/array/linalg.py]: We probably need an `x.ndim > 2` check for the Frobenius Norm as well.

File: dask/array/tests/test_linalg.py [hunk 3]
```diff
@@ -730,9 +759,31 @@ def test_norm_2dim(shape, chunks, axis, norm, keepdims):
...
 
     a_r = np.linalg.norm(a, ord=norm, axis=axis, keepdims=keepdims)
     d_r = da.linalg.norm(d, ord=norm, axis=axis, keepdims=keepdims)
 
     assert_eq(a_r, d_r)
+
+
+@pytest.mark.parametrize("shape, chunks, axis", [
+    [(4, 5, 6, 7), (2, 2, 2, 2), (1,2)],
+    [(4, 5, 6, 7), (2, 2, 2, 2), (-1,-2)],
```
Comment (jakirkham) [inline dask/array/tests/test_linalg.py]: nit: would be good to have a space after `,`s.

**Comment by jakirkham** [comment]
Thanks @TAdeJong. Took a quick look. This seems fine to me.

File: dask/array/tests/test_linalg.py (hunk 3 above)
Comment (TAdeJong) [reply dask/array/tests/test_linalg.py]: I agree. Even somewhat surprised flake8 did not pick this up

Comment (jakirkham) [reply dask/array/tests/test_linalg.py]: Yeah, also thought that was weird. @mrocklin, do you have any ideas why flake8 missed this?

**Comment by mrocklin** [comment]
It looks like we ignore this in setup.cfg

**Comment by TAdeJong** [comment]
@mrocklin: Is this fast enough to you, although we are still up from the 5ish seconds master needs on my machine?

**Comment by mrocklin** [comment]
> @mrocklin: Is this fast enough to you, although we are still up from the 5ish seconds master needs on my machine?

Should be fine.  I've marked it as slow so that it doesn't get run by default when devs run tests, but will still run on travis-ci.

File: dask/array/tests/test_linalg.py [hunk 4]
```diff
@@ -730,9 +760,31 @@ def test_norm_2dim(shape, chunks, axis, norm, keepdims):
 
     # Need one chunk on last dimension for svd.
     if norm == "nuc" or norm == 2 or norm == -2:
-        d = d.rechunk((d.chunks[0], d.shape[1]))
+        rechunkaxis = -1
+        d = d.rechunk({rechunkaxis: -1})
```
Comment (jakirkham) [inline dask/array/tests/test_linalg.py]: Maybe we should consolidate these lines?

Comment (jakirkham) [reply dask/array/tests/test_linalg.py]: Took the liberty of making this change. Hope that is ok.

**Comment by mrocklin** [comment]
OK, merging this in.  Thanks @TAdeJong for finding and resolving this and thanks @jakirkham for the review.

**Pull request merged**